substrate/frame/revive/src/lib.rs

```diff
@@ -535,21 +535,125 @@ pub mod pallet {
 	#[pallet::storage]
 	pub(crate) type OriginalAccount<T: Config> = StorageMap<_, Identity, H160, AccountId32>;
 
+	pub mod genesis {
+		use super::*;
+
+		/// Genesis configuration for contract-specific data.
+		#[derive(Clone, Default, serde::Serialize, serde::Deserialize)]
+		pub struct ContractData {
+			/// Contract code.
+			pub code: Vec<u8>,
+			/// Initial storage entries as 32-byte key/value pairs.
+			pub storage: alloc::collections::BTreeMap<[u8; 32], [u8; 32]>,
+		}
+
+		/// Genesis configuration for a contract account.
+		#[derive(Default, Clone, serde::Serialize, serde::Deserialize)]
+		pub struct Account {
+			/// Contrat address.
+			pub address: H160,
+			/// Contract balance.
+			#[serde(default)]
+			pub balance: U256,
+			/// Account nonce
+			#[serde(default)]
+			pub nonce: u32,
+			/// Contract-specific data (code and storage). None for EOAs.
+			#[serde(flatten, skip_serializing_if = "Option::is_none")]
+			pub contract_data: Option<ContractData>,
+		}
+	}
+
 	#[pallet::genesis_config]
 	#[derive(frame_support::DefaultNoBound)]
 	pub struct GenesisConfig<T: Config> {
 		/// Genesis mapped accounts
 		pub mapped_accounts: Vec<T::AccountId>,
```

Comment:
Do we still need this? I guess for accounts that are in the `pallet_balances` genesis? Can we add some docs here why we have both entries?